src/Microsoft.ML.StandardLearners/Standard/SdcaBinary.cs

```diff
@@ -120,13 +120,12 @@ private protected RoleMappedData PrepareDataFromTrainingExamples(IChannel ch, Ro
             var roles = examples.Schema.GetColumnRoleNames();
             var examplesToFeedTrain = new RoleMappedData(idvToFeedTrain, roles);
 
-            ch.AssertValue(examplesToFeedTrain.Schema.Label);
-            ch.AssertValue(examplesToFeedTrain.Schema.Feature);
-            if (examples.Schema.Weight != null)
-                ch.AssertValue(examplesToFeedTrain.Schema.Weight);
+            ch.Assert(examplesToFeedTrain.Schema.Label.HasValue);
+            ch.Assert(examplesToFeedTrain.Schema.Feature.HasValue);
+            if (examples.Schema.Weight.HasValue)
+                ch.Assert(examplesToFeedTrain.Schema.Weight.HasValue);
```

Comment:
did you mean to assert the Value :

ch.Assert(examplesToFeedTrain.Schema.Weight.Value);

Reply:
I absolutely did not mean to check on the value. First of all, I cannot, since `Nullable<T>` is a struct and `Schema.Column` is a struct, we cannot call `AssertValue` on it. Also, even if hypothetically I could check on `Value`, if `HasValue` were false then accessing that `Value` property would itself throw an exception, which is not what I want (see [documentation of `Nullable<T> here](https://docs.microsoft.com/en-us/dotnet/api/system.nullable-1.value?view=netstandard-2.0)).

If you're asking if these should be `Check` and not `Assert`, it may not be obvious but `examplesToFeedTrain` is derived from `examples` (see the forematter of this, all that stuff about shuffling and whatnot), and the roles of `examples` were already checked or otherwise confirmed to be present (see, for example, lines 102 for label, 103 for float, and line 125 in this block for weights). So asserts are the appropriate choice here, since the intention of this code is to communicate to the reader what we *expect* to be true, versus input validation, which would be more properly handled via checks or conditions (as in fact it is).